bee-network/src/peers/manager.rs

```diff
@@ -0,0 +1,564 @@
+
+            Err(e)
+        }
+        Ok(_) => {
+            // Inform the user that the command succeeded.
+            event_sender
+                .send(Event::PeerRemoved { id })
+                .map_err(|_| Error::EventSendFailure("PeerRemoved"))?;
+
+            Ok(())
+        }
+    }
+}
+
+async fn connect_peer(
+    id: PeerId,
+    local_keys: identity::Keypair,
+    peers: PeerList,
+    banned_addrs: BannedAddrList,
+    banned_peers: BannedPeerList,
+    internal_event_sender: InternalEventSender,
+    event_sender: EventSender,
+) -> Result<(), Error> {
+    // Try to reach the peer by its known ID.
+    if let Err(e) = conns::dial_peer(
+        &id,
+        &local_keys,
+        &internal_event_sender,
+        &peers,
+        &banned_addrs,
+        &banned_peers,
+    )
+    .await
+    .map_err(Error::ConnectFailure)
+    {
+        // Inform the user that the command failed.
+        event_sender
+            .send(Event::CommandFailed {
+                command: Command::ConnectPeer { id },
+            })
+            .map_err(|_| Error::EventSendFailure("CommandFailed"))?;
+
+        return Err(e);
+    }
+
+    Ok(())
+}
+
+async fn disconnect_peer(id: PeerId, peers: &PeerList, event_sender: &EventSender) -> Result<(), Error> {
+    match peers.update_state(&id, PeerState::Disconnected).await {
+        Err(e) => {
+            // Inform the user that the command failed.
+            event_sender
+                .send(Event::CommandFailed {
+                    command: Command::DisconnectPeer { id },
+                })
+                .map_err(|_| Error::EventSendFailure("CommandFailed"))?;
+
+            Err(e)
+        }
+        Ok(()) => {
+            // Inform the user that the command succeeded.
+            event_sender
+                .send(Event::PeerDisconnected { id })
+                .map_err(|_| Error::EventSendFailure("PeerDisconnected"))?;
+
+            Ok(())
+        }
+    }
+}
+
+async fn dial_address(
+    address: Multiaddr,
+    local_keys: identity::Keypair,
+    peers: PeerList,
+    banned_addrs: BannedAddrList,
+    banned_peers: BannedPeerList,
+    internal_event_sender: InternalEventSender,
+    event_sender: EventSender,
+) -> Result<(), Error> {
+    // Try to reach a peer by its known address.
+    if let Err(e) = conns::dial_address(
+        &address,
+        &local_keys,
+        &internal_event_sender,
+        &peers,
+        &banned_addrs,
+        &banned_peers,
+    )
+    .await
+    .map_err(Error::ConnectFailure)
+    {
+        // Inform the user that the command failed.
+        event_sender
+            .send(Event::CommandFailed {
+                command: Command::DialAddress { address },
+            })
+            .map_err(|_| Error::EventSendFailure("CommandFailed"))?;
+
+        return Err(e);
+    }
+
+    Ok(())
+}
+
+#[inline]
+async fn send_message(message: Vec<u8>, to: &PeerId, peers: &PeerList) -> Result<(), Error> {
```

Comment:
Seems like strange function that's more difficult to read, if anything, than the code it wraps.

Reply:
sure, you're right. The reason is this strange need for duality. There is a `recv_message` function below, so.... How strongly to you want me to change that? I feel like a lot of readers like to a `send_message` function when there is also a `recv_message` function, but I could be wrong.